src/NUnitFramework/tests/Constraints/DictionaryContainsKeyConstraintTests.cs

```diff
@@ -84,35 +85,521 @@ public void WorksWithNonGenericDictionary()
         }
 #endif
 
-        [Test]
+        [Test, Obsolete]
         public void IgnoreCaseIsHonored()
         {
             var dictionary = new Dictionary<string, string> { { "Hello", "World" }, { "Hola", "Mundo" } };
 
             Assert.That(dictionary, new DictionaryContainsKeyConstraint("HELLO").IgnoreCase);
         }
 
-        [Test]
+        [Test, Obsolete]
         public void UsingIsHonored()
         {
             var dictionary = new Dictionary<string, string> { { "Hello", "World" }, { "Hola", "Mundo" } };
 
-            Assert.That(dictionary,
-                new DictionaryContainsKeyConstraint("HELLO").Using<string>((x, y) => StringUtil.Compare(x, y, true)));
+            Assert.That(dictionary, new DictionaryContainsKeyConstraint("HELLO").Using<string>((x, y) => StringUtil.Compare(x, y, true)));
         }
 
-        [Test, Explicit("Demonstrates issue #2837")]
+        [Test]
         public void SucceedsWhenKeyIsPresentWhenDictionaryUsingCustomComparer()
         {
             var dictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Hello", "World" }, { "Hola", "Mundo" } };
 
             Assert.That(dictionary, new DictionaryContainsKeyConstraint("hello"));
         }
-        [Test, Explicit("Demonstrates issue #2837")]
+
+        [Test]
         public void SucceedsWhenKeyIsPresentUsingContainKeyWhenDictionaryUsingCustomComparer()
         {
             var dictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Hello", "World" }, { "Hola", "Mundo" } };
             Assert.That(dictionary, Does.ContainKey("hola"));
         }
+
+        [Test]
+        public void SucceedsWhenKeyIsPresentUsingContainKeyWhenUsingLookupCustomComparer()
```

Comment:
Typo, `ContainKey` should be `ContainsKey` (6 places)

These are really excellent tests, by the way!

Reply:
> These are really excellent tests, by the way!

Completely agree! 😄